xt/aspell.t

```diff
@@ -33,40 +36,81 @@ if $proc.exitcode {
 }
 
 # generate a combined words file
-my $dict = open "xt/aspell.pws", :w;
-$dict.say('personal_ws-1.1 en 0 utf-8');
-$dict.say("xt/words.pws".IO.slurp.chomp);
-$dict.say("xt/code.pws".IO.slurp.chomp);
+# a header is required, but is supplied by words.pws
+
+my $dict = $*PROGRAM.parent.child("aspell.pws").open(:w);
+$dict.say($*PROGRAM.parent.child("words.pws").IO.slurp.chomp);
+$dict.say($*PROGRAM.parent.child("code.pws").IO.slurp.chomp);
 $dict.close;
 
 my %output;
 
-sub test-it($promises, $file) {
-    await Promise.allof: |$promises;
-    my $tasks = $promises».result;
+my $lock = Lock.new;
+
+my @jobs = @files.race.map: -> $file {
+    # We use either the raw markdown or the rendered/cached Pod.
+    my $input-file = $file.ends-with('.pod6') ?? Pod::Cache.cache-file($file) !! $file;
+
+    # split the input file into a block of code and a block of text
+    # anything with a leading space is considered code, and we just
+    # concat all the code and text into one block of each per file
+
+    my Str $code = '';
+    my Str $text = '';
 
-    my $count;
-    for %output{$file}.lines -> $line {
-        FIRST next; # dump first line
-        next if $line eq '';
-        diag $line;
-        $count++;
+    # Process the text so that aspell understands it.
+    # Every line starts with a ^
+    # turn \n and \t into spaces to avoid \nFoo being read as "nFoo" by aspell
+    for $input-file.IO.slurp.lines -> $line {
+        my Bool $is-code = $line.starts-with(' ');
+
+        my $processed =  '^' ~ $line.subst(/ \S <( '\\' <[tn]> )>/ , ' ', :g) ~ "\n";
+
+        $code ~= $processed if $is-code;
+        $text ~= $processed unless $is-code;
     }
 
-    my $so-many  = $count // "no";
-    ok !$count, "$file has $so-many spelling errors";
+    for <code text> -> $type {
+        # Restrict dictionary used based on block type
+        my ($dict, $body);
+        if $type eq "code" {
+            $body = $code;
+            $dict = $*PROGRAM.parent.child("aspell.pws").absolute;
+        } else {
+            $body = $text;
+            $dict = $*PROGRAM.parent.child("words.pws").absolute;
+        }
+
+        react {
+            my $proc = Proc::Async.new(:w, ['aspell', '-a', '-l', 'en_US', '--ignore-case', "--extra-dicts=$dict", '--mode=url']);
+
+            whenever $proc.stdout.lines {
+                $lock.protect: {
+                    %output{$file}{$type}<output> = (%output{$file}{$type}<output> // '') ~ $_ ~ "\n";
```

Comment:
Won't auto-vivification take care of this?
```
> my %h
{}
> %h<foo> ~= 'hehe'
hehe
> %h
{foo => hehe}
```
Just append `"$_\n"`, no?

Reply:
f6d325843